roll/__init__.py

```diff
@@ -109,26 +109,36 @@ class Request(dict):
 
     The default parsing is made by `httptools.HttpRequestParser`.
     """
-    __slots__ = ('url', 'path', 'query_string', 'query', 'method', 'body',
-                 'headers', 'route', '_cookies')
+    __slots__ = ('app', 'url', 'path', 'query_string', '_query', 'method',
+                 'body', 'headers', 'route', '_cookies')
 
-    def __init__(self):
+    def __init__(self, app):
+        self.app = app
```

Comment:
This allows to keep `Query` reference on `Roll`.
Other option is to have `Query` on `Request` itself, but that means overriding `Request` AND `Query` when we'd only need to override `Query`.